elfpy/math/fixed_point.py

```diff
@@ -372,6 +372,45 @@ def __ceil__(self) -> FixedPoint:
                 return FixedPoint(str(int(lhs) + 1) + ".0")  # increase integer component by one
         return FixedPoint(lhs + ".0")  # the number has no remainder
 
+    def __round__(self, n: int = 0) -> FixedPoint:
+        """Returns a number rounded following Python `round` behavior.
+
+        Given a real number x and an optional integer n, return as output the number rounded to the closest multiple of 10 to the power -n.
+        If n is omitted, it defaults to 0 (round to nearest integer). Uses Python's "round half to even" strategy.
+        """
+        if not self.is_finite():
+            return self
+        lhs, rhs = str(self).split(".")  # lhs = integer part, rhs = fractional part
+        if n >= len(rhs):
+            # If n is larger than the number of decimal places, return the number itself.
+            return self
+        # Check the digit at the nth decimal place
+        digit = int(rhs[n])
+        if n == 0 or len(rhs) < n:
+            left_digit = int(lhs[-1])
+        else:
+            left_digit = int(rhs[n - 1])
+        # If these conditions are met, we should round down
+        if digit < 5 or (  # digit less than 5 OR
+            digit == 5  # digit is exactly 5 AND
+            and all(d == "0" for d in rhs[n + 1 :])  # all of the following digits are zero AND
+            and (left_digit % 2 == 0)  # the digit to the left is even
```

Comment:
why does this matter?  it seems crazy to me that you'd round 1.75 to 1.8 and 1.45 to 1.4.  Is that really the standard?

Reply:
Yeah that's how python does rounding (round to even) https://realpython.com/python-rounding/#pythons-built-in-round-function

Reply:
we can add args later to implement other methods: https://realpython.com/python-rounding/#a-menagerie-of-methods